docs/concepts.md

```diff
@@ -67,6 +68,96 @@ that are specific to a particular Node (like per-process UNIX file descriptors),
 so handle value 3 for Node A identify a different channel than handle value 3
 for Node B (unless there happens to be a coincidence of numbering).
 
+## Labels
+
+Labels are used to enforce information flow control (IFC) between nodes and
+channels in an Oak Application and the outside world.
+
+Each node and each channel is associated with a label at creation time. The
+label does not change any more after that (apart from explicit declassification
+operations, which are not currently supported).
+
+A label has two components: **secrecy** and **integrity**.
+
+Each component is represented as an (unordered) set of tags, and each tag refers
+to individual security principal in the system.
+
+The following types of security principals are supported in Oak:
+
+- **user**: specified by a bearer token, public key, or some other assertion of
```

Comment:
Are these genuinely the only two types of principals supported in Oak or are these examples? I would expect these to just be examples. I would imagine another useful common case would be principals that represent resources, for example a data store, the network, a chat room handle, etc. 

Separately, I do like that users are associated with fixed, atomic principals. This seems like a good way to ensure there is no hidden downgrading at the boundaries of the system until the system becomes truly distributed.

Reply:
They are the ones that are currently (or will shortly be) supported in Oak :) Definitely we can add more in the future, but I would not want the documentation to project itself too much in the future, so I'll keep it close to the actual implementation.

The examples you proposed sound really interesting!

I'm not sure how they would fit in Oak though, in what sense would a principal refer to the network, for instance? Would it be used as a "more precise" principal, instead of "public", to mean that it is actually "public via the network, but not in any other way"? Is that actually useful in terms of privacy for users? As far as they know, public is public, whether it's via the network or, say, via a printer. But I suspect I might be missing your point.

Reply:
Your point about wanting the documentation to reflect the current implementation totally makes sense, so do keep the text as-is.

As for considering resources as principals in the future:

"The network" might not be a great example of a principal that represents a resource since "public" will probably work just fine. I think I agree with you that "public is public", though maybe some Oak app writers really would value differentiating information release through the network or through a printer specifically. Maybe the app writer is more confident that the printer is just emitting secrets in a particular building (on a piece of paper) unlike the network (??) I agree this is not a perfect example. 

Maybe a proprietary algorithm is an example of a "resource principal"? Maybe in a medical information sharing scenario, it is useful to have a separate principal just for a patient's records (rather than using the principal of either the doctor or the patient or some combination of the doctor's principal and the patient's)? 

At a high level "resources as principals" is just something that appears to come up enough information flow projects that it seems worth considering including.